### PR TITLE
extending the list of supported publish-options

### PR DESCRIPTION
**Pull request by DenysGonchar**
subj

### CODE REVIEW

**Comment by mongoose-im** [comment]
[8032.1](https://travis-ci.org/esl/MongooseIM/jobs/665668302) / Erlang 22.0 / small_tests / 958ee69db4bda8706f3897b3b3165164c7e4c6d9
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2681/8032/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2681/8032/small_tests.22.0/small/ct_run.mongooseim@localhost.2020-03-23_01.24.29/lib.mongooseim.logs/run.2020-03-23_01.24.29/suite.log.html)

File: doc/user-guide/push-notifications/Push-notifications-client-side.md [hunk 1]
```diff
@@ -112,25 +112,33 @@ To enable push notifications in the simplest configuration, just send the follow
...
-  * `topic` - currently only used with _APNS_. The value is passed to _APNS_ as `topic` header.  For
+  * `topic` - currently only used with _APNS_. The value is passed to _APNS_ as `topic` header. For
     more information please refer to _APNS_ documentation.
   * `silent` - if set to `true`, all notifications will be "silent". This means that only the data
     payload will be send to the push notifications provider with no notification. The data payload
     will contain all notification fields as defined in [XEP-0357].
   * `priority` — which may be either `normal` or `high`, and if not given, defaults to `normal`.
     This value will set the push notification priority. Please refer to FCM / APNS documentation for
     more details on those values.
+  * `sound` - sound that should be played when notification arrives. Please refer to _FCM_/_APNS_ documentation for more details.
```
Comment (NelsonVides) [inline doc/user-guide/push-notifications/Push-notifications-client-side.md]:
```suggestion
  * `sound` - sound that should be played when a notification arrives. Please refer to _FCM_/_APNS_ documentation for more details.
```

File: doc/user-guide/push-notifications/Push-notifications-client-side.md [hunk 2]
```diff
@@ -112,25 +112,33 @@ To enable push notifications in the simplest configuration, just send the follow
...
+  * `mutable_content` - only applicable to _APNS_. If set to `true`, sets "mutable-content=1" in _APNS_ payload.
```
Comment (NelsonVides) [inline doc/user-guide/push-notifications/Push-notifications-client-side.md]:
```suggestion
  * `mutable_content` - only applicable to _APNS_. If set to `true`, sets "mutable-content=1" in the _APNS_ payload.
```

File: doc/user-guide/push-notifications/Push-notifications-client-side.md [hunk 3]
```diff
@@ -112,25 +112,33 @@ To enable push notifications in the simplest configuration, just send the follow
...
+  * `time_to_live` - only applicable to _FCM_. Maximum lifespan of an FCM notification. Please refer to _FCM_ documentation for more details.
```
Comment (NelsonVides) [inline doc/user-guide/push-notifications/Push-notifications-client-side.md]:
```suggestion
  * `time_to_live` - only applicable to _FCM_. Maximum lifespan of an FCM notification. Please refer to the _FCM_ documentation for more details.
```

File: doc/modules/mod_push_service_mongoosepush.md [hunk 4]
```diff
@@ -1,21 +1,7 @@
...
- * `topic` - _null_ if not supplied
- * `service` - has to be specified and the value must be valid and supported by the MongoosePush push service provider.
-  E.g. `fcm`, `apns`.
- * `device_id` - has to be specified and the value must be valid device token received from push notification service provider specified in `service` option
-
-In addition to those `publish-options` you may also specify a `silent` option,
-which, when set to `true` will result in "silent" notification.
-Silent notifications send only data payload to push notifications service provider with all
-fields specified in the notification without any modification and/or filtering.
+Each `push_notification` hook is converted as a `REST` API call to the [MongoosePush](https://github.com/esl/MongoosePush) service. You can find the full list of supported `publish-options` [here](../user-guide/push-notifications/Push-notifications-client-side.md#publish-options).
```
Comment (michalwski) [inline doc/modules/mod_push_service_mongoosepush.md]:
```suggestion
Each `push_notification` hook is converted as a `REST` API call to the [MongoosePush](https://github.com/esl/MongoosePush) service.
You can find the full list of supported `publish-options` [here](../user-guide/push-notifications/Push-notifications-client-side.md#publish-options).
```

File: doc/user-guide/push-notifications/Push-notifications-client-side.md [hunk 5]
```diff
@@ -112,25 +112,33 @@ To enable push notifications in the simplest configuration, just send the follow
...
-Those two options are the only ones required, but there are some others that are optional:
+_APNS_ or _FCM_. Please note that `publish-options` are specific to various XMPP Push Services.
+
+### Publish options
+For [mod_push_service_mongoosepush][] the next `publish-options` are mandatory:
+
+  * `device_id` - device token (here: `your_pns_device_token`) that you received from your push notification service provider (as described in [Registering with Push Service provider](#registering-with-a-push-service-provider))
+  * `service` - push notification service provider name (`apns` or `fcm`)
+
+there are also some other `publish-options` supported:
```
Comment (michalwski) [inline doc/user-guide/push-notifications/Push-notifications-client-side.md]: I think, it'd be good to mention that other options are simply ignored (by current `mod_push_service_mongoosepush`) and will never make it to the MongoosePush.

**Comment by mongoose-im** [comment]
[8038.1](https://travis-ci.org/esl/MongooseIM/jobs/665982783) / Erlang 22.0 / small_tests / 908ecf7a934fc3aa09f0084812cddbfdc1efd038
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2681/8038/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2681/8038/small_tests.22.0/small/ct_run.mongooseim@localhost.2020-03-23_17.43.29/lib.mongooseim.logs/run.2020-03-23_17.43.29/suite.log.html)

**Comment by mongoose-im** [comment]
[8040.1](https://travis-ci.org/esl/MongooseIM/jobs/665989378) / Erlang 22.0 / small_tests / 892190b2fa029c60d05848dc6e8daf0dc01ad04f
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2681/8040/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2681/8040/small_tests.22.0/small/ct_run.mongooseim@localhost.2020-03-23_18.11.51/lib.mongooseim.logs/run.2020-03-23_18.11.52/suite.log.html)

**Pull request merged**